alg/CMakeLists.txt

```diff
@@ -71,8 +71,9 @@ if (GDAL_USE_OPENCL)
   target_sources(alg PRIVATE gdalwarpkernel_opencl.h gdalwarpkernel_opencl.cpp)
 endif ()
 
-target_include_directories(alg PRIVATE ${PROJ_INCLUDE_DIRS})
-gdal_add_private_link_libraries(PROJ::PROJ)
+target_include_directories(alg PRIVATE $<TARGET_PROPERTY:PROJ::proj,INTERFACE_INCLUDE_DIRECTORIES>)
+target_compile_definitions(alg PRIVATE $<TARGET_PROPERTY:PROJ::proj,INTERFACE_COMPILE_DEFINITIONS>)
+gdal_add_private_link_libraries(PROJ::proj)
```

Comment:
Why are the extra `target_..` needed (here and in other spots)? 
A key point of imported target is to carry such transititive usage requiremens just by linking them.

Reply:
> Why are the extra target_.. needed (here and in other spots)?
> A key point of imported target is to carry such transititive usage requiremens just by linking them.

If I remember well my past experiments, that's related to the fact that "alg" is an object library and that our minimum CMake is 3.10 and that there are restrictions on versions < 3.12 or 3.13 related to the linking of an object library.

For example with 3.10, if I do:
target_link_libraries(alg PRIVATE PROJ::proj)

I get
CMake Error at alg/CMakeLists.txt:78 (target_link_libraries):
  Object library target "alg" may not link to anything.

whereas it works with 3.16

But using the gdal_target_link_libraries() function used in drivers actually solves this here for this non-driver component
gdal_target_link_libraries(TARGET alg LIBRARIES PROJ::proj)

I'll switch to that